examples/sampling-slice-adaptive-hyperrectangles-mcmc.ipynb

```diff
@@ -0,0 +1,948 @@
+      "text/plain": [
+       "<matplotlib.figure.Figure at 0x10d575c88>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import os\n",
+    "os.chdir(\"../\")\n",
+    "import pints\n",
+    "import pints.toy\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "# Create log pdf\n",
+    "log_pdf = pints.toy.GaussianLogPDF([2, 4], [[1, 0], [0, 3]])\n",
+    "\n",
+    "# Contour plot of pdf\n",
+    "levels = np.linspace(-3,12,20)\n",
+    "num_points = 100\n",
+    "x = np.linspace(-1, 5, num_points)\n",
+    "y = np.linspace(-0, 8, num_points)\n",
+    "X, Y = np.meshgrid(x, y)\n",
+    "Z = np.zeros(X.shape)\n",
+    "Z = np.exp([[log_pdf([i, j]) for i in x] for j in y])\n",
+    "plt.contour(X, Y, Z)\n",
+    "plt.xlabel('x')\n",
+    "plt.ylabel('y')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we set up a sampling routine using Slice Sampling with Adaptive Hyperrectangles."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Choose starting point for 3 mcmc chains\n",
+    "xs = [\n",
+    "    [2, 4],\n",
+    "    [3, 3],\n",
+    "    [5, 4],\n",
+    "]\n",
+    "\n",
+    "# Create mcmc routine\n",
+    "mcmc = pints.MCMCController(log_pdf, 3, xs, method=pints.SliceHyperrectanglesMCMC)\n",
+    "\n",
+    "# Add stopping criterion\n",
+    "mcmc.set_max_iterations(2000)\n",
```

Comment:
Do we really need 2000 iterations for a unimodal Gaussian? Keep in mind these notebooks should run quickly for users, and are tested regularly, so the faster the better!